src/prpy/base/mobilebase.py

```diff
@@ -85,20 +85,17 @@ def Forward(self, meters, execute=True, direction=None, **kw_args):
         if abs(numpy.linalg.norm(direction) - 1.0) > 1e-3:
             raise ValueError('Direction must be a unit vector.')
 
-        if self.simulated:
-            with self.robot.GetEnv():
-                start_pose = self.robot.GetTransform()
-                offset_pose = numpy.eye(4)
-                offset_pose[0:3, 3] = meters * direction
-                goal_pose = numpy.dot(start_pose, offset_pose)
-
-            path = create_affine_trajectory(self.robot, [ start_pose, goal_pose ])
-            if execute:
-                return self.robot.ExecutePath(path, **kw_args)
-            else:
-                return path
+        with self.robot.GetEnv():
+            start_pose = self.robot.GetTransform()
+            offset_pose = numpy.eye(4)
+            offset_pose[0:3, 3] = meters * direction
+            goal_pose = numpy.dot(start_pose, offset_pose)
+            
+        path = create_affine_trajectory(self.robot, [ start_pose, goal_pose ])
+        if execute:
+            return self.robot.ExecutePath(path, **kw_args)
         else:
-            raise NotImplementedError('DriveForward is not implemented')
+            return path
```

Comment:
I'm not up-to-date on the recent changes to `MobileBase`. I agree with whatever @jeking04 says. :smile: